src/Task/CopyEnvConfigs.php

```diff
@@ -136,9 +139,17 @@ private function determinePaths(array $packages): array
             }
         }
 
+        $envsRaw = $this->config->envConfigs();
+        $envs = [];
+        foreach ($envsRaw as $env) {
+            if (is_string($env) && $env) {
+                $envs[] = $env;
+            }
+        }
+
         $sourceFiles = [];
         foreach ($sourcesDirs as $sourceDir) {
-            foreach (['all', 'development', 'staging', 'production'] as $env) {
+            foreach ($envs as $env) {
                 $sourceFiles[$env] = $this->filesystem->normalizePath("{$sourceDir}/{$env}.php");
```

Comment:
I feel like we should have some sanitification/sanitization on the env names here.

When they where hardcoded that was not ncecessary, but I don't like having to pass arbitrary non-empty strings to `require_once`.

The first thing it comes into my mind is that using a series of `../../` coul create security issues.

Reply:
Yep, I agree.

What do you think about using of regular expression `^[a-z][a-z0-9_]*$` in previous loop?
https://github.com/inpsyde/vip-composer-plugin/blob/975a49a99cb3dd432694e5cd2ff8a3b403165289/src/Task/CopyEnvConfigs.php#L145

Reply:
I think it would be better to first build the full path, and then use something like `is_file()` + [`validate_file`](https://developer.wordpress.org/reference/functions/validate_file/).

Better treat paaths as paths, not strings.

Reply:
I was thinking in a context of "environment name" and not "path",  however, I agree that "path" approach is more solid.

I'll take a look. Thanks.